frontend/src/lib/k8s/api/v2/hooks.ts

```diff
@@ -231,9 +241,11 @@ export function useKubeObjectList<T extends KubeObjectClass>({
           cluster,
         }
       ).then(it => it.json());
-      list.items = list.items.map(
-        item => new kubeObjectClass({ ...item, kind: list.kind.replace('List', '') })
-      );
+      list.items = list.items.map(item => {
+        const itm = new kubeObjectClass({ ...item, kind: list.kind.replace('List', '') });
+        itm.cluster = cluster;
```

Comment:
Why is the cluster not set in the constructor? Is this the reason you added a cluster setter? I am asking because it feels like the cluster should be a readonly in that class.

Reply:
Yeah... I had to do this because the typing is broken for that code unfortunately. It doesn't know that there is a constructor that takes an extra argument and complains.

I guess we can look back at this once the other typing fixes are merged.